src/lib/x509/x509_ext.h

```diff
@@ -500,6 +500,76 @@ class OCSP_NoCheck final : public Certificate_Extension {
       void decode_inner(const std::vector<uint8_t>&) override;
 };
 
+/**
+* TNAuthList extension
+*
+* RFC8226 Secure Telephone Identity Credentials
+*   https://www.rfc-editor.org/rfc/rfc8226#section-9
+*/
+class BOTAN_PUBLIC_API(3, 5) TNAuthList final : public Certificate_Extension {
+   public:
+      class Entry final : public ASN1_Object {
+         public:
+            /* TNEntry choice values
+             * see: https://datatracker.ietf.org/doc/html/rfc8226#section-9 */
+            enum Type { TN_ServiceProviderCode = 0, TN_TelephoneNumberRange = 1, TN_TelephoneNumber = 2 };
+
+            struct TelephoneNumberRange {
+                  ASN1_String start;  //TelephoneNumber (IA5String)
+                  size_t count;       //2..MAX
+            };
+
+            static bool valid_TelephoneNumber(const ASN1_String& tn);
```

Comment:
No need to expose this here. Just a free-standing helper function in the .cpp file should be enough.
Namely:

```C++
namespace {

bool is_valid_telephone_number(...);

}
```

Reply:
* renamed to `is_valid_telephone_number`
* moved to the top of the `x509_ext.cpp`